freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/model/common/Status.java

```diff
@@ -33,10 +41,10 @@
     public static final Collection<Status> AVAILABLE_STATUSES = List.of(AVAILABLE, MAINTENANCE_MODE_ENABLED);
 
     public static final Collection<Status> REMOVABLE_STATUSES = List.of(AVAILABLE, UPDATE_FAILED, CREATE_FAILED, DELETE_FAILED,
-            DELETE_COMPLETED, STOPPED, START_FAILED, STOP_FAILED);
+            DELETE_COMPLETED, STOPPED, START_FAILED, STOP_FAILED, REPAIR_FAILED, UPSCALE_FAILED, DOWNSCALE_FAILED);
```

Comment:
should `UPSCALE_COMPLETED`, `DOWNSCALE_COMPLETED`, and `REPAIR_COMPLETED` also be removable? I'm not sure what `isRemovable` is used for.

Reply:
If it is removable status, we will delete the freeipa stack from in memory cache. I am reverting this part of code change as we can leave it in cache.

        if (newStatus.isRemovableStatus()) {
            InMemoryStateStore.deleteStack(stack.getId());
        }

Reply:
All new `FAILED`, `COMPLETED` statuses should be added to `REMOVABLE_STATUSES` as this indicates these are final states, no flow should be running for these stack and they can be removed from the `InMemoryStateStore`.

Reply:
Thanks for the suggestion! Now I added all the status to REMOVABLE_STATUSES. Since I have removed "repair completed", I use statuschecker to update it to available status.

Reply:
I added the newly added failed states (REPAIR_FAILED, UPSCALE_FAILED, DOWNSCALE_FAILED) to removable states now. Thanks!